src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -563,14 +556,12 @@ protected override bool LastStatementOrHasReturnStatementInReturnableConstruct()
         }
 
         protected override SyntaxToken CreateIdentifier(string name)
-            => Identifier(name);
+            => name.ToIdentifierToken();
```

Comment:
prior code was incorrect and wasn't using the helper that escapes keyword names.  fixed this and added test.